### PR TITLE
mrc-1505: basic package infrastructure

### PR DESCRIPTION
**Pull request by richfitz**
This PR adds the basics that we need to start getting a package pulled together - core dependencies, docker image, travis support (tests) and buildkite support (docker)

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Package: mintr
+Title: API for 'mint'
```
Comment (hillalex) [inline DESCRIPTION:2]: I guess it's "MINT" really?

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,2 @@
+YEAR: 2020
```
Comment (hillalex) [inline LICENSE:1]: what does this license grant? why not MIT?

File: docker/push [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+[ ! -z $(docker images -q $TAG_SHA) ] || docker pull $TAG_SHA
+
+docker tag $TAG_SHA $TAG_BRANCH
+docker push $TAG_BRANCH
+
+if [ $GIT_BRANCH == "master" ]; then
+    docker tag $TAG_SHA $TAG_LATEST
+    docker tag $TAG_SHA $TAG_VERSION
+    docker push $TAG_VERSION
+    docker push $TAG_LATEST
```
Comment (hillalex) [inline docker/push:16]: what purpose does this latest tag serve, isn't it just equivalent to master?

File: LICENSE (hunk 2 above)
Comment (richfitz) [reply LICENSE:1]: This is the weird way that R does licencing :-/

File: docker/push (hunk 3 above)
Comment (richfitz) [reply docker/push:16]: it means that `docker pull mrcide/mintr` will work, as `latest` is the standard docker convention as `master` is on git

File: LICENSE (hunk 2 above)
Comment (richfitz) [reply LICENSE:1]: I've added a full licence though for clarity as this is a common pain point...

**Pull request merged**